chain/network/src/peer_manager/network_state/tier1.rs

```diff
@@ -374,4 +375,26 @@ impl super::NetworkState {
         }
         None
     }
+
+    /// Finds a TIER1 connection for the given AccountId. Currently used only for OptimisticBlock,
+    /// which is implemented as a PeerMessage but has targets identified by AccountId.
+    /// TODO(saketh): consider simplifying things by changing the message type of OptimisticBlock.
+    pub fn get_tier1_proxy_for_account_id(
+        &self,
+        account_id: &AccountId,
+    ) -> Option<Arc<connection::Connection>> {
```

Comment:
We have almost exactly the same block in `send_message_to_account` function in chain/network/src/peer_manager/network_state/mod.rs

Might we want to merge/reuse this function there?

Reply:
They are very subtly different and I would prefer to limit blast radius of this PR. As mentioned in other comments we can revisit the exact handling of OptimisticBlock in the future and possibly convert it to a routed message. In that case we can delete this function.